### PR TITLE
Hello rails react review

### PR DESCRIPTION
**Pull request by akezeth**
- Created a new Rails app called 'hello-rails-react'.
- Installed needed node dependencies.
- Created a babel.config.js in the root folder of the project.
- Updated webpack.config.js to include the babel loader.
- Created a table for storing messages.
- Created an API endpoint that selects a random greeting from the table.
- Created a static view that will be the root of app.
- Created the App component with react-router and rendered it in static view.
- Created the Greeting component that will display a greeting. Set it up as a route in App component.
- Created a store, an action, and a reducer that will connect to API endpoint to get the random greeting.
- Display the random greeting in Greeting component.

### CODE REVIEW

File: app/javascript/components/Greeting.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchGreetings } from '../redux/greetings/greetingsSlice';
+
+const Greeting = () => {
+	const dispatch = useDispatch();
```
Comment (Qoosim) [inline app/javascript/components/Greeting.js:6]: - [ ] Great job so far. However, there's spelling error with this file name. This file name was named `grettingsSlice.js` but was referenced here as `greetingsSlice.js`. Kindly correct that.

### Spelling mistake
greetings-error

File: app/javascript/redux/store.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { configureStore } from '@reduxjs/toolkit';
+import messagesreducer from './greetings/greetingsSlice';
```
Comment (Qoosim) [inline app/javascript/redux/store.js:2]: - [ ] Same thing here too :point_down: 


store-err

File: config/routes.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Rails.application.routes.draw do
+  get 'greetings/index'
+  root 'root#index'
+end
```
Comment (Qoosim) [inline config/routes.rb]: - [ ] You might need to look into this route too. This route only generates this message alone, not random messages I created.

### For `http://127.0.0.1:3000/greetings/index`
msg-1

### For `http://127.0.0.1:3000/greetings`

msg-2

### For `http://127.0.0.1:3000`

loading

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @akezeth, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Implemented some project requirements. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @akezeth,

## APPROVED 🟢

## Highlights
✔️ Project met all technical requirements

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**